### ISSUE TITLE
`button-has-type` is too strict about permitted values

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_lint#1392]
### Lint Name

`button-has-type`

### Code Snippet

```ts
const foo = () => 'submit';

return <button
autofocus
type={foo()}
>

</button>
```

### Expected Result

This should not be marked. Type checking can check that the return value of `foo()` is actually submit. The linter should just mark if `type` is missing entirely.

### Actual Result

```
error[button-has-type]: Incorrect value for button's `type` attribute
 --> /home/lucacasonato/projects/github.com/denoland/deno/test.tsx:5:6
  | 
5 | type={foo()}
  |      ^^^^^^^
  = hint: The value of the `type` attribute must be one of `button`, `submit` or `reset`

  docs: https://lint.deno.land/rules/button-has-type


Found 1 problem
Checked 1 file
```

### Version

2.1.7+e475749

**Comment by lucacasonato** [comment]
I understand this lint, but in all honesty it is a bit flawed without type access. The following is perfectly fine, but the linter complains about it:

```tsx
function Button(props: { type: "submit" }) {
  return <button {...props}>Foo</button>;
}
```
